ArmPkg/Drivers/CpuDxe/CpuDxe.c

```diff
@@ -327,15 +327,27 @@ CpuDxeInitialize (
   //
   RemapUnusedMemoryNx ();
 
+  // MU_CHANGE [START]: Only install Memory Attribute Protocol if policy is enabled
   Status = gBS->InstallMultipleProtocolInterfaces (
                   &mCpuHandle,
                   &gEfiCpuArchProtocolGuid,
                   &mCpu,
-                  &gEfiMemoryAttributeProtocolGuid,
-                  &mMemoryAttribute,
+                  // &gEfiMemoryAttributeProtocolGuid,
+                  // &mMemoryAttribute,
                   NULL
                   );
 
+  if (gDxeMps.InstallMemoryAttributeProtocol) {
+    Status = gBS->InstallMultipleProtocolInterfaces (
+                    &mCpuHandle,
+                    &gEfiMemoryAttributeProtocolGuid,
+                    &mMemoryAttribute,
+                    NULL
+                    );
+  }

```

Comment:
We already weren't checking Status after this call, but perhaps we should be in both cases?

Reply:
In the first case, it would be fatal, in the second case, we could just log it (maybe assert, too, for debugging).

Reply:
Updated to assert on the first install, and print a DEBUG_ERROR message on the second